File: Result/cpp/ConfigMgr.cpp
#include "stdafx.h"
#include "ConfigMgr.h"

static ConfigMgr * mInstance = NULL;

ConfigMgr::ConfigMgr()
{
}


ConfigMgr::~ConfigMgr()
{
}


ConfigMgr * ConfigMgr::GetInstance() 
{
    if (NULL == mInstance)
    {
        mInstance = new ConfigMgr();
        mInstance->Init();
    }
    return mInstance;
}

void ConfigMgr::Init()
{
	ItemItem.Load();
	MsgErrcode.Load();
	MsgMsg.Load();
}

File: misc/demo/java/xlsxtools/src/com/xlsxtools/ConfigMgr.java
package com.xlsxtools;

public class ConfigMgr 
{
	private static ConfigMgr mInstance = null;
	public static ConfigMgr  GetInstance() 
	{ 
		if(null == mInstance)
		{
			mInstance = new ConfigMgr();
			mInstance.Init(); 
		}
		return mInstance;
	}
	public ItemItemConfigTable	ItemItem = null;
	public MsgErrcodeConfigTable	MsgErrcode = null;
	public MsgMsgConfigTable	MsgMsg = null;

	public void Init () 
	{
		try
		{
			ItemItem = new ItemItemConfigTable();
			ItemItem.load();
			MsgErrcode = new MsgErrcodeConfigTable();
			MsgErrcode.load();
			MsgMsg = new MsgMsgConfigTable();
			MsgMsg.load();
		}
		catch(Exception ex)
		{
			ex.printStackTrace();
		}
	}
}